### PR TITLE
ci: run jaffle shop for act1

### PR DESCRIPTION
**Pull request by isidentical**
snowflake

### CODE REVIEW

File: .github/workflows/test_dbt_fal.yml [hunk 1]
```diff
@@ -0,0 +1,167 @@
+name: Test dbt-fal adapter
+
+on:
+  pull_request:
+    types: [assigned, opened, synchronize, reopened]
```
Comment (chamini2) [inline .github/workflows/test_dbt_fal.yml:5]:
```suggestion
    types: [assigned, opened, synchronize, reopened]
    branches: ['act1**']
```

File: .github/workflows/test_dbt_fal.yml [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+      - uses: actions/checkout@v2
+        with:
+          repository: fal-ai/dbt-core
+          ref: act1
+          path: 'dbt-core'
+
+      - uses: actions/checkout@v2
+        with:
+          repository: isidentical/jaffle_shop_with_fal # change this to fal-ai/jaffle_shop_with_fal
+          ref: add-profiles-for-fal-ci  # change this to act1
```
Comment (chamini2) [inline .github/workflows/test_dbt_fal.yml:52]: TODO

File: .github/workflows/test_dbt_fal.yml [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+          # the existing revision.
+
+          pip install dbt-${{ matrix.profile }} -c fal/adapter/compatible_adapters.txt
+
+          pushd dbt-core
+          pip install -r dev-requirements.txt -r editable-requirements.txt
+          popd
+
+          pushd fal
+          pip install ".[${{ matrix.profile }}]"
```
Comment (chamini2) [inline .github/workflows/test_dbt_fal.yml:87]: I think we don't need extras here, since we are taking care database read/write in the `fal/adapter` code.

File: pyproject.toml [hunk 4]
```diff
@@ -43,7 +42,7 @@ firebase-admin = "^5.1.0"
 # Adapters
 
 ## snowflake
-snowflake-connector-python = { version = "2.7.9", optional = true } # 2.7.10 needs pyarrow 8
+snowflake-connector-python = { version = "<2.8", optional = true }
```
Comment (chamini2) [inline pyproject.toml:45]: if we really do not need the fal extras installed, we can revert this change

File: .github/workflows/test_dbt_fal.yml (hunk 1 above)
Comment (isidentical) [reply .github/workflows/test_dbt_fal.yml:5]: I thought it would do that by default  (e.g. only work on branches that inherit from `act1`)

Comment (chamini2) [reply .github/workflows/test_dbt_fal.yml:5]: you mean because of the `push` instruction? that is isolated from this one

File: .github/workflows/test_dbt_fal.yml [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+  run:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        profile: ["postgres", "bigquery", "snowflake", "duckdb"]
+        python: ["3.8"]
+        environment: ["local", "venv", "conda"]
+        # bigquery implementation fails due to concurrency, let's
+        # limit it to 1 for now.
```
Comment (chamini2) [inline .github/workflows/test_dbt_fal.yml:26]: Thinking about it: this probably will fail for snowflake too. Want to limit it?

File: .github/workflows/test_dbt_fal.yml (hunk 1 above)
Comment (chamini2) [reply .github/workflows/test_dbt_fal.yml:5]: Well, I guess that is true.

File: .github/workflows/test_dbt_fal.yml (hunk 5 above)
Comment (isidentical) [reply .github/workflows/test_dbt_fal.yml:26]: It works for now, not sure what's going on but we can limit.

Comment (isidentical) [reply .github/workflows/test_dbt_fal.yml]: I've tried without any concurrency, let's see whether it would work now (instead of running venv, local, conda parallel; now running them sequentially one by one in the same job)

File: adapter/pyproject.toml [hunk 6]
```diff
@@ -12,9 +12,28 @@ python = ">=3.7.1, <3.11"
 dbt-core = "1.3.0b2"
 fal = ">=0.6.0"
 
+## snowflake
+snowflake-connector-python = { version = "^2.7.12", extras = ["pandas"], optional = true }
+
+## bigquery
+# We need the 1.5.0 version to support passing our own client.
```
Comment (chamini2) [inline adapter/pyproject.toml:19]: ATL: should it be 1.5 then? what are we doing about this?

Comment (isidentical) [reply adapter/pyproject.toml:19]: Just to keep it here I guess, but yeah it doesn't work out of the box. I tried to add a git dependency, but seems like poetry doesn't work when you mark something as both coming from git and optional.

**Comment by isidentical** [comment]
Our solution seems to be not working for dbt-snowflake==1.0.x, needs investigation 🤔
